### PR TITLE
Add extract-api step for the PR CI build

### PR DESCRIPTION
**Pull request by Yato333**
Currently, some packages uses api-extractor tool, which helps to check for breaking API changes.
Added a build step that will fail if `pnpm extract-api` produces a result that differs from the one that is committed.

### CODE REVIEW

File: packages/itwin/tree-widget/api/tree-widget-react.api.md [hunk 1]
```diff
@@ -451,7 +453,7 @@ export interface TreeContextMenuProps {
 export interface TreeDefinition {
     getLabel: () => string;
     id: string;
-    render: () => React.ReactNode;
+    render: (props: TreeRenderProps) => React.ReactNode;
```
Comment (grigasp) [inline packages/itwin/tree-widget/api/tree-widget-react.api.md:456]: @saskliutas @jasdom, looks like we missed a breaking change. Can props be made optional?

Comment (saskliutas) [reply packages/itwin/tree-widget/api/tree-widget-react.api.md:456]: I don't think this is a breaking change. These props are always provided by `TreeSelector` when rendering selected tree. https://github.com/iTwin/viewer-components-react/blob/9d296a06dd2e2290364692e02b109e9ac591a5e9/packages/itwin/tree-widget/src/components/TreeSelector.tsx#L64

This callback should never be called by consumer. And consumers can keep passing render callback that does not need props:
```ts
 {
  id: CategoriesTreeComponent.id,
  getLabel: CategoriesTreeComponent.getLabel,
  render: () => <CategoriesTreeComponent />,
},
```

Comment (grigasp) [reply packages/itwin/tree-widget/api/tree-widget-react.api.md:456]: Agreed.

File: .pipelines/templates/build-steps.yaml [hunk 2]
```diff
@@ -20,5 +20,8 @@ steps:
   # - script: pnpm exec prettier . --check
   #   displayName: Check formatting
 
-  - script: pnpm exec lage lint cover --grouped
-    displayName: Build, Lint and Cover
+  - script: pnpm exec lage lint cover extract-api --grouped
+    displayName: Build, Lint, Cover and check for unexpected API changes
+    env:
+      # Fail the build if package uses api-extractor and there are uncommitted API changes
+      TF_BUILD: 1
```
Comment (saskliutas) [inline .pipelines/templates/build-steps.yaml:27]: Is this needed? It should be predefined (https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#system-variables-devops-services)

**Comment by saskliutas** [comment]
I think CI fails because `extract-api` task is missing dependency on `build` https://github.com/iTwin/viewer-components-react/blob/9d296a06dd2e2290364692e02b109e9ac591a5e9/lage.config.js#L34

File: .pipelines/templates/build-steps.yaml (hunk 2 above)
Comment (Yato333) [reply .pipelines/templates/build-steps.yaml:27]: I am not sure, I copied it from presentation repo

Comment (saskliutas) [reply .pipelines/templates/build-steps.yaml:27]: Presentation repo uses Github actions instead of Azure DevOps. So it should not be necessary in here,

Comment (Yato333) [reply .pipelines/templates/build-steps.yaml:27]: Let me check if this is the case

Comment (Yato333) [reply .pipelines/templates/build-steps.yaml]: Yes it works, thanks

**Review by saskliutas — APPROVED** [review]

File: change/@itwin-property-grid-react-6141cefa-e559-437e-85ff-3904f0fd207e.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
+  "comment": "Add extract api step",
```
Comment (aruniverse) [inline change/@itwin-property-grid-react-6141cefa-e559-437e-85ff-3904f0fd207e.json:3]:
```suggestion
  "comment": "",
```

File: change/@itwin-tree-widget-react-1de32109-0cd0-442d-8e17-ccf77b483c50.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
+  "comment": "Add extract api step",
```
Comment (aruniverse) [inline change/@itwin-tree-widget-react-1de32109-0cd0-442d-8e17-ccf77b483c50.json:3]:
```suggestion
  "comment": "",
```

**Review by aruniverse — APPROVED** [review]

File: change/@itwin-property-grid-react-6141cefa-e559-437e-85ff-3904f0fd207e.json (hunk 3 above)
Comment (saskliutas) [reply change/@itwin-property-grid-react-6141cefa-e559-437e-85ff-3904f0fd207e.json]: FYI, there is no need to make comment empty string. If type is set to `none` it won't impact `CHANGELOG.md` generation.

**Pull request merged**